Mathlib/Data/Nat/Defs.lean

```diff
@@ -831,11 +831,12 @@ def twoStepInduction {P : ℕ → Sort*} (zero : P 0) (one : P 1)
   | 1 => one
   | _ + 2 => more _ (twoStepInduction zero one more _) (twoStepInduction zero one more _)
 
-@[elab_as_elim]
+@[elab_as_elim, deprecated (since := "2024-09-04")]
```

Comment:
Please add the name of the lemma this is deprecated in favour of.

Reply:
```suggestion
@[elab_as_elim, deprecated strongRecOn (since := "2024-09-04")]
```